.github/workflows/publish-docs-staging.yaml

```diff
@@ -0,0 +1,42 @@
+name: Publish docs-staging.k8ssandra.io
+
+on:
+  push:
+    branches:
+      - 'docs'
```

Comment:
That's not what we want anymore. This workflow should trigger when we push to `main`.